### PR TITLE
doc: update release versions

### PR DESCRIPTION
**Pull request by greg-fer**
Updated release version for 1.2.0 release.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

**Edit ru-fu:**
This PR should be merged just before tagging, together with the release notes PR.

### CODE REVIEW

**Comment by greg-fer** [comment]
This is tracked in NCSDK-4385 and based on the following versions provided by @rlubos :

zephyr: 2.1.99
mcuboot: 1.4.99
nrf: 1.2.0
nrfxlib: 1.2.0 (TBC)

**Comment by greg-fer** [comment]
Checked links.txt against the contents of the repo:

1) The following links are empty in the file, and used in the documentation:

.. _`nRF Command Line Tools`:
.. _`nRF5x Command Line Tools`:
.. _`nRF5x Command Line Tools Linux 32`:
.. _`nRF5x Command Line Tools Linux 64`:
.. _`nRF5x Command Line Tools Windows 32`:
.. _`nRF5x Command Line Tools Windows 64`:

2) The following links are in the links.txt file, but missing in the documentation:
.. |ncs_zephyr_repo| replace::
.. |ncs_mcuboot_repo| replace::
.. _`GitHub duplicate`:
.. _`Battery Service Specification`:
.. _`AWS S3 Developer Guide: Using Bucket Policies and User Policies`:
.. _`AWS S3 Developer Guide: Bucket Policy Examples`:

@ru-fu , please advise what to do with them.

**Comment by ru-fu** [comment]
>     1. The following links are empty in the file, and used in the documentation:

As discussed, these should stay because they are alternative names for the link at the end.

>     1. The following links are in the links.txt file, but missing in the documentation:

If they aren't used anywhere, they should be deleted from the links.txt file.

**Comment by thst-nordic** [comment]
This CI passed 4 days ago.  
If you make a change and re-run CI it will fail, and you will need to rebase on master.    
If you want to merge it without any changes that is ok too.

File: doc/nrf/links.txt [hunk 1]
```diff
@@ -19,10 +19,8 @@
 .. |ncs_repo| replace:: https://github.com/NordicPlayground/fw-nrfconnect-nrf
 .. _`fw-nrfconnect-nrf`: https://github.com/NordicPlayground/fw-nrfconnect-nrf
 
-.. |ncs_zephyr_repo| replace:: https://github.com/NordicPlayground/fw-nrfconnect-zephyr
 .. _`fw-nrfconnect-zephyr`: https://github.com/NordicPlayground/fw-nrfconnect-zephyr
 
-.. |ncs_mcuboot_repo| replace:: https://github.com/NordicPlayground/fw-nrfconnect-mcuboot
```
Comment (rlubos) [inline doc/nrf/links.txt]: Why are these removed and others (i.e. ncs_nrfxlib_repo) left?
